test/libsolidity/semanticTests/modifiers/transient_state_variable_value_type.sol

```diff
@@ -0,0 +1,17 @@
+contract C {
+    uint16 transient x;
+
+    modifier m(uint16) {
+        x += 10;
+        _;
+    }
+
+    function f() public m(x) returns (uint16 x) {
+        x *= 10;
+    }
+}
+
+// ====
+// EVMVersion: >=cancun
+// ----
+// f() -> 0
```

Comment:
It does - `returns (uint16 x)`.

Reply:
Ah, true. I did not notice that both the state variable and the return variable are called `x` :)

I'd still say that the test is broken in a way. Is that naming on purpose? The way it's written, you'll never get anything other than `0` no, matter if `x` is resolved to the right variable or not. You should call the return variable something else or just use `return`.

Reply:
I mean, you 'wrote' the [test](https://github.com/ethereum/solidity/pull/15292#discussion_r1769029146). But you are right, we even discussed this particular test case (because I was confused with the test's behaviour), and it ended up being due to shadowing, which I forgot to change.

Reply:
But you broke it :P It's the shadowing that's problematic and that wasn't in my version :)

Reply:
Wait, it didn't? The only difference (aside from removing the unused address variable) is specify the return parameter?

Reply:
It wasn't. I mean my example was incomplete and you obviously had to add something to be able to inspect the value of `x`. But it did not have that shadowing :)